test/StringProcessing/Sema/regex_literal_type_inference.swift

```diff
@@ -0,0 +1,12 @@
+// RUN: %target-typecheck-verify-swift -enable-experimental-string-processing
+
```

Comment:
I *think* you may need to add `REQUIRES: libswift` to this test

Reply:
I didn't seem to need it to run any of the string processing tests. Do you know if it's required? I actually get UNSUPPORTED when it does `REQUIRES: libswift`, but the test passes when I drop it.

Reply:
Hmm yeah I'm not entirely sure as I was mainly going off the existing tests. I *think* it's still possible to build the compiler with libswift disabled though, though I don't know if any of the bots use that configuration, cc @eeckstein

Reply:
Erik says it's needed so I'm adding it back, but I'll still need to figure out why I'm getting UNSUPPORTED when running locally.

Reply:
>  I don't know if any of the bots use that configuration

Windows builds without libswift.

> Erik says it's needed

It's needed for tests depending on libswift. Is this the case here? There are no changes to libswift in this PR.

Reply:
All tests that contain regex literals need libswift because the parser uses it.